library/std/src/sys/thread_local/os.rs

```diff
@@ -133,13 +220,23 @@ unsafe extern "C" fn destroy_value<T: 'static>(ptr: *mut u8) {
     // Note that to prevent an infinite loop we reset it back to null right
     // before we return from the destructor ourselves.
     abort_on_dtor_unwind(|| {
-        let ptr = unsafe { Box::from_raw(ptr as *mut Value<T>) };
```

Comment:
Similarly to the above thought: is it plausible that for aligned thread locals we could generate a repr(C) struct that bumped the alignment of our stored value to what we need and then we'd alter our code to `&struct.0` when actually returning pointers out to userspace? Presumably that would avoid the majority of the internal changes and avoid storing alignments into memory (wastefully).